### PR TITLE
Add confirmation message window for Block option in API life cycle tab.

### PR DESCRIPTION
**Pull request by nimanthag**
Signed-off-by: nimanthag <nimanthag@wso2.com>

## Purpose
Resolves https://github.com/wso2/product-apim/issues/3625

## Goals
Add confirmation message window for Block option in API life-cycle tab. 

## Approach
the confirmation window will pop up with following text once they click on "Block" button.
text = "Are you sure you want to block this API ?"
image

If the user clicks on "yes" it will execute the Block functionality otherwise no.

## User stories
N/A

## Release note
N/A
## Documentation
N/A

## Training
N/A

## Certification
N/A

## Marketing
N/A

## Automation tests
N/A

## Security checks
N/A

## Samples
N/A

## Related PRs
N/A

## Migrations (if applicable)
N/A

## Test environment
N/A

## Learning
N/A

### CODE REVIEW

**Comment by bhathiya** [comment]
Is this fix only for the block option?

**Comment by bhathiya** [comment]
That's fine. Shall we check all possible state changes and add confirmation messages to which don't have one already?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag [hunk 1]
```diff
@@ -187,7 +187,73 @@ function updateStatus(status) {
                 $(".modal-footer").html("");
             }
         });
-    } else {
+    } else if (status == 'Block'){
```
Comment (mushthaq33) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]: There is a formatting issue

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag [hunk 2]
```diff
@@ -187,7 +187,73 @@ function updateStatus(status) {
...
+        var messageContent = i18n.t("Are you sure you want to block this API ?");
```
Comment (isharac) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]: No space is required after the 'API'

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag [hunk 3]
```diff
@@ -187,7 +187,73 @@ function updateStatus(status) {
...
+                        if (result.status.stateChangeStatus == 'REJECTED') {
+                            showWorkflowRejectedMessage();
+                        } else if (jsonPayload != null && jsonPayload != "") {
+                            var apiInfo = {};
+                            apiInfo.provider = provider;
+                            apiInfo.name = apiName;
+                            apiInfo.version = version;
+                            handleWorkflowRedirection(jsonPayload, apiInfo);
+                        } else {
+                            ///test
```
Comment (vithu30) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]: shall we remove this comment.

**Comment by nimanthag** [comment]
> That's fine. Shall we check all possible state changes and add confirmation messages to which don't have one already?

Sure. Text messages for each transition mention below
1) Are you sure you want to publish this API?
2) Are you sure you want to block this API?
3) Are you sure you want to demote to created this API?
4) Are you sure you want to retire this API?
5) Are you sure you want to deploy as a prototype this API? 
6) Are you sure you want to re-publish this API?
7) Once the API is deprecated; you can only retire the API as the next step.Are you sure you want to deprecate the API") + " - " + apiName + " - " + version + "?"
8) All the previous versions of the API will be deprecated. Continue?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag (hunk 3 above)
Comment (nimanthag) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]: Sure. I will remove this.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag (hunk 2 above)
Comment (nimanthag) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]: Sure. I will update this as requested.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag (hunk 1 above)
Comment (nimanthag) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]: I will correct this in next commit.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag [hunk 4]
```diff
@@ -46,156 +46,56 @@ function setChecklistItemVal(element,order) {
...
-                $('#messageModal').modal('hide');
-                $(".modal-footer").html("");
-            }
-        });
+        messageContent = i18n.t("All the previous versions of the API will be deprecated. Continue?");
     } else {
-        jagg.post("/site/blocks/life-cycles/ajax/life-cycles.jag", {
+        var displayStatus = status.toLowerCase();
+        if (status == 'Demote to Created') {
+        displayStatus = displayStatus.replace("to created", ""); // Remove "to created" part
```
Comment (bhathiya) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]: formatting error

Comment (nimanthag) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]: corrected

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag [hunk 5]
```diff
@@ -46,156 +46,56 @@ function setChecklistItemVal(element,order) {
...
-                $('#messageModal').modal('hide');
-                $(".modal-footer").html("");
-            }
-        });
+        messageContent = i18n.t("All the previous versions of the API will be deprecated. Continue?");
     } else {
-        jagg.post("/site/blocks/life-cycles/ajax/life-cycles.jag", {
+        var displayStatus = status.toLowerCase();
+        if (status == 'Demote to Created') {
+            displayStatus = displayStatus.replace("to created", ""); // Remove "to created" part
```
Comment (bhathiya) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]:
```suggestion
            messageContent = i18n.t("Are you sure you want to demote this API to created state?");
```

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag [hunk 6]
```diff
@@ -46,156 +46,56 @@ function setChecklistItemVal(element,order) {
...
+        }
+        messageContent = i18n.t("Are you sure you want to ") + displayStatus + " this API?";
```
Comment (bhathiya) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]: making only a part of the message i18n is not useful. Shall we make the entire message i18n?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag [hunk 7]
```diff
@@ -46,156 +46,56 @@ function setChecklistItemVal(element,order) {
...
-                $(".modal-footer").html("");
-            }
-        });
+        messageContent = i18n.t("All the previous versions of the API will be deprecated. Continue?");
     } else {
-        jagg.post("/site/blocks/life-cycles/ajax/life-cycles.jag", {
+        var displayStatus = status.toLowerCase();
+        if (status == 'Demote to Created') {
+            displayStatus = displayStatus.replace("to created", ""); // Remove "to created" part
+        }
```
Comment (bhathiya) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]:
```suggestion
        } else {
```

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag (hunk 6 above)
Comment (nimanthag) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]: In an early implementation,
messageContent = i18n.t("Once the API is deprecated; you can only retire the API as the next step.Are you sure you want to deprecate the API") + " - " + apiName + " - " + version + "?";
part of the message is taken to i18n and append variables to it.

Comment (nimanthag) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]: Complete the change as requested.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag [hunk 8]
```diff
@@ -46,156 +46,56 @@ function setChecklistItemVal(element,order) {
...
-            cancelCallback: function() {
-                $('#messageModal').modal('hide');
-                $(".modal-footer").html("");
-            }
-        });
+        messageContent = i18n.t("All the previous versions of the API will be deprecated. Continue?");
     } else {
-        jagg.post("/site/blocks/life-cycles/ajax/life-cycles.jag", {
+        if (status == 'Demote to Created') {
+            messageContent = i18n.t("Are you sure you want to demote this API to created state?");
```
Comment (bhathiya) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/life-cycles/template.jag]:
```suggestion
            messageContent = i18n.t("Are you sure you want to demote this API to CREATED state?");
```

**Pull request merged**